crates/typst-syntax/src/parser.rs

```diff
@@ -539,8 +534,10 @@ fn math_args(p: &mut Parser) {
             ")" => break,
             ";" => {
                 maybe_wrap_in_math(p, arg, named);
-                p.wrap(array, SyntaxKind::Array);
-                p.convert(SyntaxKind::Semicolon);
+                // Must use wrap_within to include wrapping the empty math element when
+                // trivia is present.
+                p.wrap_within(array, p.marker(), SyntaxKind::Array);
```

Comment:
is this and the below a bug fix? if yes, it would be good to have a regression test.

Reply:
Yes? I'm not fully sure why these were needed tbh, but it fixed an issue in the math-call tests (that *I* added :upside_down_face: ). I really hate this function in particular and I'm planning a separate math-parsing PR for later anyways (3 weeks?).